[조수연] Payment 미션 Step2 (한번 더 리팩토링!...ing)

suyeon1218 opened the pull request:
# 구현하면서 어려웠던 점

- Primitive UI 를 제대로 이해하지 못했다는 아쉬움이 남아 한번더 리팩토링을 진행했습니다! 우선 cardList 페이지와 cardRegist 페이지만 리팩토링 하여 올려두었습니다! ㅠㅠ... 면접일정이 모두 끝나고 나면 다시 재도전 해야겠어요. 
- [배포링크](https://react-payments-custom-zeta.vercel.app)

### 컴포넌트 구조 변경

- [FSD 아키텍쳐](https://emewjin.github.io/feature-sliced-design/)를 적용했습니다.
- page, slice, layer 의 3 depth 로 이뤄져 있는데 ui 같은 경우 layer 에서도 추가 depth가 필요하지 않나 싶어서... 확신은 없지만 일단 사용해 보았습니다!
- `widgets`, `features` 의 역할이 특히 헷갈려서 파일을 여기저기로 옮겨보는 시도를 한다고 도중에 시간을 많이 잡아 먹었습니다. 조금 더 자세히 공부할 필요가 있다고 생각합니다... ㅠㅠ
- 제가 잡아둔 체계라서 이해가 가는 건진 모르겠지만... 어떤 ui나 util 함수를 가져올 때 '이건 어느 폴더에 있겠다' 라는 생각이 바로바로 들었던 것 같아요. 

### Primitive UI

- 기능을 제공하고 스타일은 외부에서 부여하는 형식으로 컴포넌트를 재작성 했습니다.
- 그런데 Headless UI와는 어떤 차이점이 있는지 잘 몰라서... 이것도 더 공부할 영역입니다.
- ark ui 스타일과 chakra ui 를 조금 섞어서 만들어 보려고 했어요. 구조같은 건 chakra를 참조하고 `Component.Root` 와 같은 이름은 ark ui의 네이밍에서 따왔습니다.
- 이전이랑 비교했을 때 크게 변한 게 없는 것 같은데도 컴포넌트를 유연하게 잘 사용했다는 착각(?) 이 들었어요.
- 원래는 `PropsWithChildren` 과 `CSSProperties` 를 모두 일일이 작성했는데 `ComponentProps` 하나면 정리가 다 돼서 편리했습니다.
- 더불어 react-hook-form 도 뭔가 좀 더 사용하기 편리해진 느낌...? 이 듭니다. 저번보다 react-hook-form 라이브러리의 삽질을 훨씬 절 하게 됐어요. 

# 궁금한 점 & 리뷰받고 싶은 점

### FSD 아키텍쳐

- 이건 제가 공부를 더 해야하는 부분이지만... entities 에선 도메인 데이터를 담당하는데, 도메인 데이터를 어떻게 분리하는지 감이 잘 안와요. 저희 프로젝트에선 도메인 데이터가 `CreditCard` 만 있는 게 맞을까요?

- `features` 폴더는 사용자의 비즈니스 시나리오를 담당합니다! 예시로 댓글이 있다면 댓글 쓰기, 댓글 삭제 등이 이에 해당한다고 해요. 저희 프로젝트에선 카드 등록 (추가) / 카드 관리 (삭제, 편집) 으로 나눠봤는데... 이 시나리오로 나누는 게 맞는지 궁금합니다. ㅠㅠ

suyeon1218 committed:
feat: 카드 만료일 Input 기능 구현

suyeon1218 committed:
feat: InputWrapper에 div 엘리먼트 props 추가하기

suyeon1218 committed:
style: Modal에 relative 속성 추가하기

suyeon1218 committed:
feat: 비밀번호 입력 컴포넌트 구현

suyeon1218 committed:
style: Information Header 문자 크기 조정하기

suyeon1218 committed:
style: Information border스타일을 Payments로 옮기기

suyeon1218 committed:
feat: 카드 선택 컴포넌트 구현하기

suyeon1218 committed:
chore: 카드 회사 상수로 분리하기

suyeon1218 committed:
refactoring: 카드 타입을 회사로 분류하기

suyeon1218 committed:
style: Information 컴포넌트 패딩값 없애기

suyeon1218 committed:
feat: 카드 회사 입력받는 컴포넌트 구현하기

suyeon1218 committed:
refactoring: Card 컴포넌트 위치 변경하기

suyeon1218 committed:
feat: CardAlias UI 구현

suyeon1218 committed:
style:전체 스타일 조정

suyeon1218 committed:
feat: 카드 등록시 유효성 검사 추가하기

suyeon1218 committed:
feat: 카드 추가하는 기능 구현하기

suyeon1218 committed:
feat: 카드 삭제 기능 구현

suyeon1218 committed:
feat: 각 페이지에서 뒤로가기 기능 구현

suyeon1218 committed:
feat: InputPassword 유효성 검사가 값 입력 초기에도 이뤄지도록 하기

suyeon1218 committed:
feat: Payments 페이지 form 유효성 검사 뒤 결제하기 활성화하기

suyeon1218 committed:
feat: 카드 선택시 카드 이름 변경 창으로 이동하는 기능 구현

suyeon1218 committed:
feat: CardNumber 사이에 대시 넣기

suyeon1218 committed:
chore: import-order ESLint 적용

suyeon1218 committed:
refactoring: Homepage 리팩토링

- 함수 분리하기

suyeon1218 committed:
Refactoring: Payment 페이지 구조 리팩토링하기

- useState와 useEffect없이 Button렌더링하기
- payments로 사용하던 경로를 payment로 변경하기
- 하위 컴포넌트 이름 미리 작성해두기

suyeon1218 committed:
chore: 경로 별칭 세팅하기

suyeon1218 committed:
refactoring: 경로 별칭 ESLint 적용하기

suyeon1218 committed:
refactoring: CardSelection 리팩토링

- PayCardInformation -> CardSelction으로 이름 변경
- TitleText 컴포넌트 분리
- 경로 별칭 적용
- import-order 적용

suyeon1218 committed:
refactoring: Card Refactoring

- clickable props를 다른 props를 통해서 유추하기
- 보안된 카드 번호들 변수명 변경하기

suyeon1218 committed:
refactoring: CardSelector -> CompanySelection 이름 변경

suyeon1218 committed:
refactoring: PaymentAmount 리팩토링

- 결제 금액을 다양하게 받기 위해 외부에서 주입받기
- style파일 분리하기
- PayAmountInformation -> PaymentAmount로 이름 변경

suyeon1218 committed:
refactoring: TermAgreement 리팩토링

- style 파일 분리하기
- PayTermInformation -> TermAgreement 로 이름 변경하기

suyeon1218 committed:
refactoring: CardSlide 리팩토링 하기

- Card에 onClick메서드를 직접 넣는 방식으로 수정
- onClick메서드 함수로 분리하기

suyeon1218 committed:
refactoring: CardList 페이지 리팩토링 하기

- 이벤트 위임으로 메서드 관리하기
- Card 타입 분리하기

suyeon1218 committed:
refactoring: CardAlias 페이지 리팩토링

- 버튼 메서드 분리하기
- import-order ESLint 적용하기

suyeon1218 committed:
refactoring: CompanySelection 컴포넌트 리팩토링

- DeemBackground 컴포넌트를 별도의 컴포넌트로 분리

suyeon1218 committed:
refactoring: InputCompany 컴포넌트 리팩토링

- DeemBackground 와 CompanySelection 분리하기
- validate 추가하기
- style 컴포넌트 분리하기
- InputCard -> InputCompany로 파일 이름 변경하기

suyeon1218 committed:
refactoring: HTMLAttributes 대신 ComponentProps 타입 사용하기

suyeon1218 committed:
feat: number가 string타입이 아닌경우 타입가드 작성하기

suyeon1218 committed:
refactoring: InputCardNumbers 컴포넌트 리팩토링

- InputNumbers -> InputCardNumbers로 파일 이름 변경
- InputRef를 배열로 관리하기
- Controller를 map으로 반복하기
- import order ESLiint 적용하기
- 경로 별칭 사용하기
- 불필요한 state 삭제하기
- TitleText를 별개의 컴포넌트로 분리하기

suyeon1218 committed:
refactoring: Input컴포넌트 리팩토링하기

- 불필요한 props를 기본 InputElement타입으로 받기

suyeon1218 committed:
refactoring: TextTitle 컴포넌트 적용하기

suyeon1218 committed:
chore: 변경된 카드 타입 적용하기

suyeon1218 committed:
refacotoring: InputEndDate 컴포넌트 리팩토링

- 불필요한 useState, useEffect 삭제
- import-order ESLint 적용
- inputRef를 배열로 관리하기
- 불필요한 rules 삭제

suyeon1218 committed:
refactoring: InputCardUser 컴포넌트 리팩토링

- import-order ESLint 적용
- 최대 20자 문구 추가

suyeon1218 committed:
refactoring: InputSecurityCode 컴포넌트 리팩토링

- 불필요한 useState, useEffect 삭제
- import-order ESLint 적용
- 스타일 파일 푼리

suyeon1218 committed:
fix: Deembackground 컴포넌트의 onClose가 제대로 동작하지 않는 문제 수정

- event.target 에 제한을 둠으로써 수정

suyeon1218 committed:
feat: Input 컴포넌트 value 타입 값을 input 엘리먼트의 defaultValue 타입과 동일하게 받기

suyeon1218 committed:
refactoring: CardRegist 페이지 리팩토링

- Keypad에서 Deembackground 분리하기
- KeypadModal 컴포넌트 분리
- ReadonlyPassword 컴포넌트 분리
- 불필요한 useState, useEffect 삭제
- import-order ESLint 적용
- 경로 별칭 적용
- useRef를 배열로 관리하기
- 이벤트 위임 방식으로 이벤트 관리하기

suyeon1218 committed:
style: 다음 버튼이 우측에 위치하도록 하기

suyeon1218 committed:
fix: 카드 비밀번호 입력이 안되는 문제 수정

suyeon1218 committed:
refactoring: CardRegist 페이지 리팩토링

- 이벤트 핸들링 메서드 따로 분리하기

suyeon1218 committed:
style: Payment 컨테이너에 너비값 주기

suyeon1218 committed:
chore: Storybook 설치

suyeon1218 committed:
feat: Button 컴포넌트 스토리북 등록하기

suyeon1218 committed:
refactoring: Button props 이름 변경하기

- invalid -> disabled

suyeon1218 committed:
refactoring: 경로 별칭 적용하기

suyeon1218 committed:
feat: 카드 타입에 none 추가하기

suyeon1218 committed:
feat: Card 컴포넌트 storybook 구현하기

suyeon1218 committed:
feat: Input 컴포넌트 storybook 구현하기

suyeon1218 committed:
feat: InputWrapper storybook 구현

suyeon1218 committed:
refactoring: InputWrapper에 import-order ESLint 적용

suyeon1218 committed:
feat: Keypad컴포넌트 storybook 연결

suyeon1218 committed:
feat: CompanySelection컴포넌트 storybook에 연결

suyeon1218 committed:
refactoring: Button스토리북 import-order ESLint 적용

suyeon1218 committed:
fix: components목록에 CompanySelection이 나타나지 않는 문제 수정

- prop을 name에서 title로 수정

suyeon1218 committed:
feat: DeemBackground컴포넌트를 스토리북에 연결

suyeon1218 committed:
refacotoring: 컴포넌트 이름 변경

- Card 컴포넌트 -> CreditCard 컴포넌트

suyeon1218 committed:
refactoring: Modal과 Infromation병합하기

suyeon1218 committed:
style: Card컴포넌트 배경색 추가

suyeon1218 committed:
feat: Card컴포넌트 스토리북 등록

suyeon1218 committed:
refactoring: PrevIcon 컴포넌트 삭제

- GoogleIcon 컴포넌트와 병합하기

suyeon1218 committed:
feat: 스토리북 preview-head 설정하기

- pretendard 폰트 링크 추가
- google Icon 링크 추가

suyeon1218 committed:
feat: GoogleIcon 스토리북 추가하기

suyeon1218 committed:
fix: HomePage 에러 수정

- 분리한 Deembackground 와 Card 컴포넌트 코드를 반영하지 않아서 생긴 문제 수정

suyeon1218 committed:
fix: Payment 페이지 수정

- Modal 컴포넌트가 Card 컴포넌트와 병합되어 발생한 문제 수정

suyeon1218 committed:
refactoring: CreditCard에서 카드를 추가하는 컴포넌트 제외시키기

suyeon1218 committed:
feat: AddCreditCard 컴포넌트 추가하기

suyeon1218 committed:
fix: CardRegist 페이지 에러 수정

- Information 컴포넌트가 Card 컴포넌트로 이름이 변경되며 생긴 문제 수정

suyeon1218 committed:
fix: CardRegist 오류 수정

- information 컴포넌트를 Card 컴포넌트로 수정하지 않아 발생한 오류 수정

suyeon1218 committed:
fix: HomePage 스타일이 어긋나있던 문제 수정

- S.ContentContainer 빠트린 것 추가하기
- Card 를 Modal 로 이름 바꿔사용하기

suyeon1218 committed:
style: CardList 모달의 높이가 작은 문제 수정

- Modal의 크기가 컨테이너의 높이에 맞게 증가할 수 있도록 변경
- ContentContainer의 배경색을 흰색으로 지정

suyeon1218 committed:
fix: CardRegist 페이지 수정

- 기존의 CreditCard 컴포넌트 대신AddCreditCard 컴포넌트를 사용하도록 변경

suyeon1218 committed:
fix: CardRegist 페이지 스타일 수정

- Keypad 모달과 CompanySelection 모달의 bottom 적용 위치가 다른 문제 수정
- ContentContainer가 두번 들어간 문제 수정
- Card가 부모 컨테이너의 크기를 넘을 경우 overflow-y 속성을 지정해서 부모의 크기를 따르도록 수정

suyeon1218 committed:
fix: 아이콘 클릭 시 click 이벤트가 동작하지 않는 문제 수정

- span element의 property를 제대로 전달해줌으로써 해결

suyeon1218 committed:
refactoring: Slide와 Card 분리하기

suyeon1218 committed:
fix: 필요한 Card 컴포넌트에만 overflow 적용하기

- Card 전체에 overflow를 함으로써 생기는 문제 해결

suyeon1218 committed:
feat: Slide 스토리북 연결

suyeon1218 committed:
refactoring: AddCreditCard 컴포넌트 내보내기 추가하기

suyeon1218 committed:
feat: AddCreditCard 컴포넌트 스토리북에 등록하기

suyeon1218 committed:
feat: 카드 번호 사이에 '-' 기호 추가하기

suyeon1218 committed:
faat: 만료일의 최소 날을 설정하기

suyeon1218 committed:
feat: 유저 이름의 최대 길이를 30자로 변경

suyeon1218 committed:
feat: 사용자 이름 입력시 현재 상태를 위에 보여주기

suyeon1218 committed:
feat: 카드 별칭 등록시 placeholder 수정

suyeon1218 committed:
feat: 별칭이 없는 경우 카드 이름으로 별칭 등록하기

suyeon1218 committed:
feat: 카드 별칭 길이 10자로 제한하기

suyeon1218 committed:
style: 모달 크기 키우기

suyeon1218 committed:
style: Payments 카드에서 Footer 와 Body 사이 간격주기

suyeon1218 committed:
style: CardList 페이지 Header 폰트 크기 키우기

suyeon1218 committed:
fix: cardUser의 초기값이 undefined 여서 길이를 가져오지 못하는 문제 수정

suyeon1218 committed:
chore: 불필요한 테스트 파일 삭제

suyeon1218 committed:
feat: 유효성 검사 로직 분리하기

suyeon1218 committed:
feat: form 제출 전 전체 값 검사하는 기능 구현

suyeon1218 committed:
chore: createValidator 파일 삭제

suyeon1218 committed:
feat: cardRegistFormValidate 유효성 검사 테스트 코드 작성

suyeon1218 committed:
feat: generateID 유틸함수 테스트 작성

suyeon1218 committed:
feat: suffle 유틸 함수에 대한 테스트 작성

suyeon1218 committed:
feat: useModal 훅 테스트 케이스 작성

suyeon1218 committed:
fix: CardColor 에러 수정

- cardColor 에 none 프로퍼티가 없어서 난 오류 수정

suyeon1218 committed:
fix: suffle 메서드 타입 수정

- suffle이 특정 개수로 이뤄진 배열을 반환하지 않을 수도 있는 에러 수정
- 특정 개수를 받아 해당 개수만큼의 배열만을 생성하는 타입 생성

suyeon1218 committed:
fix: suffle에 제네릭 전달하기

suyeon1218 committed:
fix: TypeError 해결
- suffle 메서드에 튜플을 할당하는 경우 나타나는 타입 에러 수정

seeyoujeong commented on src/components/AddCreditCard/index.tsx:
```diff
@@ -0,0 +1,24 @@
+import * as S from './index.style';
+import GoogleIcon from '../GoogleIcon';
+
+export interface AddCreditCardProps extends PropsWithChildren {
+  onClick?: () => void;
+}
+
+const AddCreditCard = ({ onClick }: AddCreditCardProps) => {
```
아래처럼 쓸 수도 있어요!

```typescript
const AddCreditCard = ({ children, onClick }: PropsWithChildren<AddCreditCardProps>) => {
```

suyeon1218 replied:
맞아요...! 이런 방법으로 쓴 컴포넌트도 몇 있는데 스타일 통일을 하는 게 좋겠네요!

seeyoujeong commented on src/components/Card/CardBody.tsx:
```diff
@@ -0,0 +1,11 @@
+
+interface CardBody extends PropsWithChildren {}
+
+const CardBody = ({ children }: CardBody) => {
```
`React.HTMLAttributes<HTMLDivElement>`라는 타입을 써서 기존 div태그 프로퍼티를 다쓸 수 있어요!

suyeon1218 replied:
이것도 어떤 컴포넌트엔 적용시켰는데 어떤 곳엔 적용을 하지 못했어요...!!

seeyoujeong commented on src/components/CompanySelection/index.tsx:
```diff
@@ -0,0 +1,48 @@
+import { cardCompany } from '~/constants';
+import { CardCompany } from '~/types';
+
+function isCardCompany(key: string): key is keyof typeof cardCompany {
+  return key in cardCompany;
+}
```
제네릭을 써서 유틸함수로 빼도 괜찮을 것 같네요. ㅎㅎ

seeyoujeong commented on src/components/CreditCard/index.tsx:
```diff
@@ -0,0 +1,62 @@
+  onClick,
+}: CreditCardProps) => {
+  const clickable = onClick ? true : false;
+  const encryptedNumbers = Array.from(numbers).map((number, index) => {
+    if (typeof number === 'string') {
+      return index < 2 ? number : '●'.repeat(number.length);
+    }
+    return number;
+  });
```
input 태그의 type에 password를 안쓰시고 함수를 만드신 이유가 있을까요??

suyeon1218 replied:
`components` 내부에 있는 컴포넌트들은 다 같은 레벨이라고 생각해서 `Input` 을 안 쓰려고 일부러 string 형식으로 표현했어요. 별로 좋은 방식이 아니라 생각했는데 찬욱님 코드를 보다보니 다음과 같은 해결 방법이 떠올랐습니당

1. component 레벨별로 분류하기
2. 순수 input 엘리먼트 가져다 쓰기 <- 이 정도는 컴포넌트 레벨을 분리하지 않아도 괜찮을 것 같단 생각이 방금 들었어요..!

seeyoujeong commented on src/components/Input/index.tsx:
```diff
@@ -0,0 +1,62 @@
+    },
+    ref,
+  ) => {
+    const [value, setState] = useState<
+      string | number | readonly string[] | undefined
+    >(defaultValue);
+
+    const handleChangeInput = (event: ChangeEvent<HTMLInputElement>) => {
+      const { target } = event;
+
+      if (target instanceof HTMLInputElement) {
+        const { maxLength } = target;
+        const newValue = event.target.value;
+
+        if (onlyNumber && isNaN(Number(newValue))) {
+          return;
+        }
+        if (!onlyNumber || (onlyNumber && !isNaN(Number(newValue)))) {
+          onChange && onChange(event);
+          setState(newValue);
+        }
+        if (maxLength === newValue.length) {
+          onComplete && onComplete(event.target);
+        }
+      }
+    };
```
이부분은 커스텀 훅으로 만들어도 좋을 것 같네요!

suyeon1218 replied:
Input 컴포넌트 안에서만 쓴다고 생각했는데 다른 곳에서도 쓰일 일이 있을까요?!

seeyoujeong replied:
아 그렇긴하죠 ㅎㅎ 만약 해당 Input 컴포넌트가 공용으로 쓰인다면 하는일이 제한적인게 아닐까라는 생각에 말씀드린건데 너무 짧게 말씀드렸네요...

seeyoujeong commented on src/pages/CardList/index.tsx:
```diff
@@ -0,0 +1,83 @@
+    navigate(`/card-alias/${id}`);
+  };
+
+  const handleClickCardUL = (event: MouseEvent<HTMLUListElement>) => {
+    const { currentTarget, target } = event;
+
+    if (currentTarget instanceof HTMLUListElement) {
+      const $li = currentTarget.querySelector('li') as HTMLLIElement;
+      const { id } = $li.dataset;
+
+      if (target instanceof HTMLButtonElement) {
+        id && handleClickRemoveButton(id);
+      }
+
+      if (target instanceof HTMLDivElement) {
+        id && handleClickRegisteredCard(id);
+      }
+    }
+  };
```
오 이벤트 위임을 쓸 생각을 못했는데..! 저도 적용해봐야겠네요! :)

seeyoujeong commented on src/utils/suffle.ts:
이렇게 작성할 수도 있어요!

```typescript
const suffle = <T>(arr: T[]): T[] => {
  const suffledArray = [...arr];

  for (let i = suffledArray.length - 1; i > 0; i--) {
    const j = Math.floor(Math.random() * (i + 1));

    [suffledArray[i], suffledArray[j]] = [suffledArray[j], suffledArray[i]];
  }

  return suffledArray;
};
```

suyeon1218 replied:
그러게요!! 괜히 extends 써서 더 복잡하게 표현했다는 느낌이 드네요...!!

seeyoujeong reviewed (commented):
코드 잘 읽었습니다! 
컴포넌트 분리를 엄청나게 하셔서 대단하다고 느꼈습니다. :)

---

> 함수 분리

저 같은 경우는 코드가 길어지면 기명함수로 만들어주고 코드가 짧으면 익명함수로 해줬네요!

> 이벤트 위임

이벤트 위임을 쓸까라는 생각도 못했었는데 쓰신걸 보고 저도 써야지라는 생각을 했습니다. ㅎㅎ
비슷한 이벤트가 여러곳에 등록되는것보단 위임을 하신게 전 좋다고 생각해요! 

> React 테스트

훅 테스트 코드 작성하는건 처음이라 막막했었는데 ai가 훅 테스트 코드 작성을 어느정도 해줘서 많은 도움을 받았어요! 덤으로 작성해준 코드를 바탕으로 공부하니까 막힘없이 공부를 할 수 있었어요!

suyeon1218 committed:
refactoring: App.tsx 파일과 관련한 파일의 폴더 구조 수정

suyeon1218 committed:
refactoring: 변경된 App 파일로 폴더 경로 변경

suyeon1218 committed:
chore: 기존 스토리북 삭제

suyeon1218 committed:
feat: Button 컴포넌트 구현

suyeon1218 committed:
refactoring: util 함수 shared 폴더로 이동하기

suyeon1218 committed:
feat: Card 컴포넌트 구현

suyeon1218 committed:
feat: Field 컴포넌트 구현하기

suyeon1218 committed:
refactoring: import 문 합치기

suyeon1218 committed:
feat: Input 기능 확장하기

- maxLength일 경우 onComplete이벤트 실행하기
- state 기반으로 관리하기
- ref 를 prop으로 받을 수 있게 하기

suyeon1218 committed:
feat: Keypad 컴포넌트 구현하기

suyeon1218 committed:
refactoring: GoogleIcon 상수 분리하기

suyeon1218 committed:
feat: GoogleIcon 컴포넌트 구현하기

suyeon1218 committed:
feat: Box Layout 구현하기

suyeon1218 committed:
refactoring: GoogleIcon 스토리북 컨벤션 수정

- 파일 이름 수정
- Default가 아닌 Root로 이름 수정

suyeon1218 committed:
refactoring: Box 를 layout 폴더에서 ui 폴더로 이동

suyeon1218 committed:
refactoring: 새로 변경한 theme 로 stotybook 적용하기

suyeon1218 committed:
style: Button 컴포넌트의 기본 스타일을 Emotion 으로 옮기기

suyeon1218 committed:
refactoring: useModal 테스트 파일을 useDisclosure 로 변경

suyeon1218 committed:
fix: 사용하지 않는 storybook dependencies 삭제하기

suyeon1218 committed:
refactoring: 기존 useModal 삭제하기

suyeon1218 committed:
refactoring: 기존 theme 삭제하기

suyeon1218 committed:
feat: googleIcon 에 닫기 아이콘 추가

suyeon1218 committed:
feat: Modal 컴포넌트 구현

suyeon1218 committed:
refactoring: Modal Root 타입을 contextValue 타입으로 확장하기

suyeon1218 committed:
chore: Modal 스타일 컴포넌트 이름 알맞게 수정

- Content -> Overlay로 수정

suyeon1218 committed:
style: Modal.Header 폰트 사이즈 증가

suyeon1218 committed:
feat: Drawer 컴포넌트 구현

suyeon1218 committed:
refactoring: PropsWithChildren 타입 삭제

suyeon1218 committed:
refactoring: Keypad 컴포넌트 리팩토링

- styled 파일로 스타일 분리
- 키패드 색상 변경

suyeon1218 committed:
refactoring: Label 컴포넌트 스타일 분리하기

suyeon1218 committed:
refactoring: Card 컴포넌트 스타일파일 분리하기

suyeon1218 committed:
refactoring: Box 컴포넌트 스타일 파일 분리하기

suyeon1218 committed:
refactoring: 리팩토링한 컴포넌트 삭제하기

suyeon1218 committed:
refactoring: 불필요한 라우팅 삭제

suyeon1218 committed:
chore: 변경된 상수 키 테마 컬러에 적용하기

suyeon1218 committed:
refactoring: CreditCard 리팩토링

suyeon1218 committed:
style: CreditCard 의 폰트 크기가 변하지 않는 문제 수정

suyeon1218 committed:
feat: CompanyItem 구현하기

suyeon1218 committed:
refactoring: CardSelctor 컴포넌트 리팩토링 하기

suyeon1218 committed:
feat: 공통 Header ui 구현

suyeon1218 committed:
style: 기본 Field 스타일 변경

- input 패딩 넓히기
- input 컬러 수정하기
- Label과 에러 메세지 폰트 크기 감소시키기

suyeon1218 committed:
chore: shared 폴더의 ui 모듈 외부로 내보내기

suyeon1218 committed:
feat:  Field 컴포넌트의 InputWrapper 컴포넌트 구현

suyeon1218 committed:
feat: Input이 styleType을 받도록 기능 추가

suyeon1218 committed:
chore: InputWrapper 내보내기 추가

suyeon1218 committed:
feat: CardUserField 컴포넌트 구현

suyeon1218 committed:
chore: 기본 Header 컴포넌트 파일 삭제

suyeon1218 committed:
refactoring: CardList Header 컴포넌트 리팩토링

suyeon1218 committed:
chore: features 폴더에서 내보내기 설정하기

suyeon1218 committed:
feat: Field.ErrorText 에 에러 값 설정하기

suyeon1218 committed:
refactoring: feature 하위 폴더 이름 변경

- RegistCard -> Form 으로 이름 변경

suyeon1218 committed:
feat: CreditCard의 크기를 선택적으로 받도록 하기

suyeon1218 committed:
chore: cardAlias 타입을 alias 로 변경

suyeon1218 committed:
chore: 이전 파일 위치 변경

suyeon1218 committed:
feat: Page 공용으로 쓰는 Layout 컴포넌트 구현

suyeon1218 committed:
feat: AddCard 컴포넌트 구현하기

suyeon1218 committed:
feat: CardList 페이지 AddCard widget 구현

suyeon1218 committed:
feat: CardList 페이지 RegistedCards 위젯 구현

suyeon1218 committed:
refactoring: CardList 페이지 리팩토링

suyeon1218 committed:
refactoring: CardRegist 페이지의 Header 리팩토링

suyeon1218 committed:
fix: GoogleIcon 에서 key값을 받아 아이콘 출력하기

suyeon1218 committed:
feat: Widgets 에 공통 레이아웃 정의하기

suyeon1218 committed:
refactoring: 이전에 사용하던 파일 삭제

suyeon1218 committed:
refactoring: CardRegist 페이지 Header UI 구현

suyeon1218 committed:
refactoring: widgets 에 있던 컴포넌트를 ui 폴더에서 관리하기

suyeon1218 committed:
feat: Layout에 Modal 기능 추가하기

suyeon1218 committed:
fix: CardList 페이지 헤더가 나타나지 않는 문제 수정

suyeon1218 committed:
refactoring: 불필요한 프로퍼티 삭제

suyeon1218 committed:
refactoring: ManageCard 페이지 리팩토링

- CardList -> ManageCard 로 이름 변경
- widget 에 존재하던 파일을 pages/ui 슬라이스로 이동

suyeon1218 committed:
feat: CardRegist 페이지 Header ui 구현하기

suyeon1218 committed:
feat: Card 상태띄워주는 ui 구현

suyeon1218 committed:
refactoring: entities 폴더에서 ui 슬라이스 내보내기

suyeon1218 committed:
refactoring: 유효성 검사 로직 features 로 이동하기

suyeon1218 committed:
feat: CompanyField 구현하기

suyeon1218 committed:
feat: CompanyUserField 구현하기

suyeon1218 committed:
chore: 유효성 검사 파일 경로 수정하기

suyeon1218 committed:
style: Field 너비 넓히기

suyeon1218 committed:
feat: 현재 카드 소유자 문자열 길이 표시하기

suyeon1218 committed:
chore: 사용하지 않는 파일 삭제하기

suyeon1218 committed:
feat: Input 필드 uncontrolled로 관리하기

suyeon1218 committed:
feat: NumberField 다음 Input으로 자동 포커스 하기

suyeon1218 committed:
fix: Number Field 유효성 검사 안 하는 문제 수정하기

suyeon1218 committed:
feat: 숫자만 입력받을 수 있도록 하기

suyeon1218 committed:
fix: 유효성 검사가 제대로 안이뤄지는 문제 수정

suyeon1218 committed:
feat: 카드 소유자 필드 Controller 로 관리하기

suyeon1218 committed:
feat: CVCField 컴포넌트 구현하기

suyeon1218 committed:
feat: PassswordField 구현하기

suyeon1218 committed:
feat: ExpirationDateField 컴포넌트 구현

suyeon1218 committed:
feat: CompanyField 를 react-hook-form 에 등록하기

suyeon1218 committed:
feat: form 다음 버튼 기능 추가

suyeon1218 committed:
refactoring: 중복되는 CSSProperties 삭제하기

suyeon1218 committed:
refactoring: testCode 의 유닛함수 경로 변경

suyeon1218 committed:
chore: cardAlias 페이지 임시 삭제

suyeon1218 committed:
chore: 사용하지 않는 엘리먼트 스타일 삭제

suyeon1218 committed:
chore: eslint 파일에 prettier 적용

suyeon1218 committed:
fix: onlyNumber Input에 숫자가 아닌 문자가 입력되는 문제 수정

suyeon1218 committed:
fix: 카드 회사 선택을 하지 않아도 다음 버튼이 나타나는 문제 수정

suyeon1218 committed:
chore: CardList -> ManageCard 로 이름 변경

suyeon1218 committed:
feat: CardAlias Header 컴포넌트 구현하기

suyeon1218 committed:
feat: Router 경로 수정

- manage-card로 경로 변경
- CardAlias 페이지 경로 추가

suyeon1218 committed:
feat: 카드 등록 폼에서 별칭 폼으로 넘어갈 때 카드 추가 디스패치 호출하기

suyeon1218 committed:
feat: InputAlias 컴포넌트 구현하기

suyeon1218 committed:
feat: 카드 리스트에 카드 정보 보여주기

suyeon1218 committed:
feat: 카드 클릭시 카드 별칭 페이지로 이동하기

suyeon1218 committed:
style: 카드 별칭 스타일 주기

suyeon1218 committed:
feat: 카드 별칭에 기본 값 추가하기

suyeon1218 committed:
fix: 루트 경로로 진입할 시 잘못된 경로로 이동하던 문제 수정

suyeon1218 committed:
fix: CardRegist 페이지에서 뒤로 가기 경로 수정

suyeon1218 committed:
fix: Input 의 onlyNumber 프로퍼티가 제대로 동작하지 않는 문제 수정

suyeon1218 committed:
fix: 만료일의 유효성 검사가 제대로 동작하지 않는 문제 수정

suyeon1218 committed:
fix: 카드사를 선택하지 않아도 다음버튼이 나타나는 문제 수정

suyeon1218 committed:
refactoring: Disclosure UI 사용법 변경하기

- onClose, onOpen 의 용도를 이벤트 용도로 변경하기
- 외부에서 초기값과 이벤트 받을 수 있도록하기
- useClosure 를 Provider 내부에서 호출하기

suyeon1218 committed:
fix: Widget에서 Modal 사용방식 변경하기

suyeon1218 committed:
fix: page에서 사용하는 모달 창 사라지는 문제 수정하기

suyeon1218 committed:
style: widget 페이지 모달 크기 수정하기

suyeon1218 committed:
style: 모달 overflow 설정

suyeon1218 committed:
style: 커서 스타일 추가하기

suyeon1218 committed:
fix: Drawer 나타나지 않는 문제 수정

suyeon1218 committed:
chore: CSSProperties 프로퍼티 삭제

suyeon1218 committed:
style: 헤더에 content가 가려지는 문제 수정

suyeon1218 committed:
fix: 카드 관리 페이지의 navigate 경로 수정

suyeon1218 committed:
chore: CompanyItem 스토리북 기본 값 추가하기

suyeon1218 committed:
fix: 변경된 스펙에 대해 테스트 수정하기

suyeon1218 committed:
chore: pnpm 대신 npm 사용해서 빌드하기

suyeon1218 committed:
Revert "chore: pnpm 대신 npm 사용해서 빌드하기"

This reverts commit 66d4dea8806e07253e2d1d364ec30f9ab20fbcba.

suyeon1218 committed:
feat: chromatic 추가하기

suyeon1218 committed:
chore: chromatic 워크 플로우 추가

suyeon1218 committed:
fix: 제대로 된 워크플로우로 수정

suyeon1218 committed:
refactoring: 스토리북의 기본 컴포넌트 이름을 Default로 변경

suyeon1218 committed:
feat: stroybook accesiibility 및 test-runner 추가하기

suyeon1218 committed:
feat: 사용자 접근성 테스트 통과시키기

- green500 컬러값 수정하기
- label 값 사용하기

suyeon1218 committed:
feat: vercel ci/cd에 choromatic 의존성 주입하기

suyeon1218 committed:
feat: ignore 브랜치 수정

suyeon1218 committed:
fix: Production CI/CD workflow로 변경하기

suyeon1218 committed:
fix: 의존성 설정하기

suyeon1218 committed:
fix: Unexpect  Token 에러 수정

suyeon1218 committed:
fix: Deploy-Production 단계에 chromatic 의존성 추가하기
